src/gax.ts

```diff
@@ -292,23 +312,147 @@ export class CallSettings {
   }
 }
 
+/**
+ * Validates passed retry options in preparation for eventual parameter deprecation
+ * converts retryRequestOptions to retryOptions
+ * then sets retryRequestOptions to null
+ *
+ * @param {CallOptions} options - a list of passed retry option
+ * @return {CallOptions} A new CallOptions object.
+ *
+ */
+export function checkRetryOptions(
+  options?: CallOptions,
+  gaxStreamingRetries?: boolean
+): CallOptions | undefined {
+  // options will be undefined if no CallOptions object is passed at call time
+  if (!options) {
+    return options;
+  }
+  // if a user provided retry AND retryRequestOptions at call time, throw an error
+  // otherwise, convert supported parameters
+  if (!gaxStreamingRetries) {
+    // if user is opted into legacy settings but has passed retry settings, let them know there might be an issue if it's a streaming call
+    if (options.retry !== undefined) {
+      warn(
+        'legacy_streaming_retry_behavior',
+        'Legacy streaming retry behavior will not honor settings passed at call time or via client configuration. Please set gaxStreamingRetries to true to utilize passed retry settings. gaxStreamingRetries behavior will be set to true by default in future releases.',
+        'DeprecationWarning'
+      );
+    }
+    if (options.retryRequestOptions !== undefined) {
+      warn(
+        'legacy_streaming_retry_request_behavior',
+        'Legacy streaming retry behavior will not honor retryRequestOptions passed at call time. Please set gaxStreamingRetries to true to utilize passed retry settings. gaxStreamingRetries behavior will convert retryRequestOptions to retry parameters by default in future releases.',
+        'DeprecationWarning'
+      );
+    }
+    return options;
+  }
+  if (
+    options.retry !== undefined &&
+    options.retryRequestOptions !== undefined
+  ) {
+    throw new Error('Only one of retry or retryRequestOptions may be set');
+  } // handle parameter conversion from retryRequestOptions to retryOptions
+  if (options.retryRequestOptions !== undefined) {
+    if (options.retryRequestOptions.objectMode !== undefined) {
```

Comment:
I'd prefer && to nested if statements, it's a bit easier to read!

Reply:
@sofisl, I think I want to push back on this - right now we have a lot that happens within the nested `if (options.retryRequestOptions !== undefined)` block because that statement must be true for the next 80 lines of code to happen. I know we have a lot of checking for if something is defined or not throughout this PR and there are definitely cases where I or someone in the past have done so in a way that isn't very pretty but this is one instance where I truly am not sure of a cleaner approach because of this first guardrail of a statement needing to be there (but I'm open to learning if you have one!)

Reply:
Hmm I'm realizing optional chaining could maybe make this better and I'll mess around with it

Reply:
Totally! I'm also ok with if-statements, just that many nested block statements becomes a bit difficult to read. Not a blocker though, more of a nit :)

Reply:
Nits are very welcome especially given that JS style evolves rather quickly and I'm still getting up to speed with it! I appreciate you :)

Reply:
Just went through these and was able to simplify a bunch with some nullish coalescing and optional chaining. Not all, but a bunch. Definitely less nesting now.